templates/model/decode/decode.go

```diff
@@ -228,3 +236,27 @@ func decodeFromVersionKind(filename, apiVersion, kind string, buf []byte) (model
 
 	return vu, nil
 }
+
+// LatestSupportedAPIVersion is the most up-to-date API version. It's
+// in the format "cli.abcxyz.dev/v1beta4".
+//
+// isReleaseBuild is the value of version.IsReleaseBuild(), but for testing
+// purposes we make it an argument rather than hardcoding.
+func LatestSupportedAPIVersion(isReleaseBuild bool) string {
+	// Release builds (like "I am the official release of version 1.2.3") will
+	// read and write only officially released, finalized api_versions. Other
+	// builds (e.g. CI builds, local dev builds, devs running "go test" on
+	// workstations) are more permissive and will read and write the most recent
+	// unreleased work-in-progress api_version.
+	if !isReleaseBuild {
+		return apiVersions[len(apiVersions)-1].apiVersion
+	}
+	for i := len(apiVersions) - 1; i > 0; i-- {
```

Comment:
optional: https://pkg.go.dev/golang.org/x/exp/slices#Reverse?

Reply:
I considered it, but it's a little awkward here because we'd have to make a copy, then reverse the copy, then iterate.